data/mods/stadium/scripts.ts

```diff
@@ -121,10 +122,17 @@ export const BattleScripts: ModdedBattleScriptsData = {
 		}
 	},
 	tryMoveHit(target, pokemon, move) {
-		let doSelfDestruct = true;
 		let damage: number | false | undefined = 0;
 
-		// First, check if the Pokémon is immune to this move.
+		// First, check if the target is semi-invulnerable
+		let hitResult = this.runEvent('Invulnerability', target, pokemon, move);
+		if (hitResult === false) {
+			if (!move.spreadHit) this.attrLastMove('[miss]');
+			this.add('-miss', pokemon);
+			return false;
+		}
+
+		// Then, check if the PokÃ©mon is immune to this move.
```

Comment:
```suggestion
		// Then, check if the Pokémon is immune to this move.
```

This is why I prefer to leave Unicode out of source code.

But also wow, what are you doing with a Unicode-incompatible text editor???

Reply:
The terminal I use with my test server has a built in text editor that I used for quick changes which seems to lack Unicode, forgot to fix the unicode afterwards